### ISSUE TITLE
setDate is not working properly

### ISSUE THREAD

**Issue by MihirPalkhiwala** [lopspower/LazyDatePicker#7]
App crashes when you set a date programmatically and tries to change value.

`lazyDatePicker.setDate(new Date());`

java.lang.NullPointerException: Attempt to invoke interface method 'void android.text.TextWatcher.beforeTextChanged(java.lang.CharSequence, int, int, int)' on a null object reference

**Comment by lopspower** [comment]
Do you use the latest version of the library? This has been corrected in this one:

```
implementation 'com.mikhaellopez:lazydatepicker:1.0.4'
```

**Comment by MihirPalkhiwala** [comment]
Yes. Using latest version 1.0.4

**Comment by lopspower** [comment]
Can you send me your XML and/or your implementation to try to helped you ? (maybe a sample project with your problem)
Because everything works well in the sample project by adding:

```java
// Init LazyDatePicker
LazyDatePicker lazyDatePicker = findViewById(R.id.lazyDatePicker);
lazyDatePicker.setDate(new Date());
```

and

```xml
<com.mikhaellopez.lazydatepicker.LazyDatePicker
        android:id="@+id/lazyDatePicker"
        android:layout_width="wrap_content"
        android:layout_height="wrap_content"
        app:ldp_date_format="mm-dd-yyyy" />
```

<img src="https://i.ibb.co/QKW5nW0/capture.png" />

**Comment by MihirPalkhiwala** [comment]
XML
`<com.mikhaellopez.lazydatepicker.LazyDatePicker
                                android:id="@+id/lazyDatePicker"
                                android:layout_gravity="center"
                                android:layout_width="match_parent"
                                android:layout_height="wrap_content"
                                app:ldp_date_format="mm-dd-yyyy"
                                android:padding="@dimen/size_5dp"
                                app:ldp_hint_color="@color/primaryGray"
                                app:ldp_text_color="@color/colorPrimary" />`


`LazyDatePicker lazyDatePicker = findViewById(R.id.lazyDatePicker);
lazyDatePicker.setDate(new Date());`


Now just try to pick a new date and it will give an error.

**Comment by lopspower** [comment]
It's now fix on the last version **1.0.5**:

```groovy
implementation 'com.mikhaellopez:lazydatepicker:1.0.5'
```

Thx for you feeback
Can you confirm it's now ok for you ?

**Comment by MihirPalkhiwala** [comment]
Fixed now. Excellent. :)
